### PR TITLE
Updates to basic-config to support MPV of Step Function Workflow

### PR DESCRIPTION
**Pull request by nabadger**
- Allow `Service` and `ExternalSecrets` to support a `name` or `nameOverride` value (we need option to name these explicitly)
- Fix `Service.selectorLabelsOverride` again

### CODE REVIEW

File: charts/basic-config/values.yaml [hunk 1]
```diff
@@ -81,13 +81,16 @@ configMaps: []
...
+#    nameoverride: ""
 #    refreshIntervalOverride: ""
 #    secretStoreRefOverride: ""
 
 # -- Define a headless Service
 service:
   # -- Determines whether to create a headless service or not
   enabled: false
+  # -- Service to use. A name is generated using the mintel_common.fullname template if it is not set
+  name: ""
```
Comment (barrymars) [inline charts/basic-config/values.yaml:93]: i'd call this nameOverride, as a name would be created if it wasn't set

File: charts/basic-config/templates/service.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 apiVersion: {{ include "common.capabilities.service.apiVersion" . }}
 kind: Service
 metadata:
-  name: {{ include "mintel_common.fullname" $serviceData }}
+  name: {{ default (include "mintel_common.fullname" $serviceData) .Values.service.name }}
```
Comment (barrymars) [inline charts/basic-config/templates/service.yaml:6]: might want to move this into a helped called something like `servicename`

and would need to be `{{ default (include "mintel_common.fullname" $serviceData) $serviceData.nameOverride }}`

will also need to check where else we are using the service name, eg: ingresses

File: charts/basic-config/templates/external-secrets.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 apiVersion: {{ include "common.capabilities.externalsecret.apiVersion" $ }}
 kind: ExternalSecret
 metadata:
-  name: {{ include "mintel_common.fullname" $secretData }}
+  name: {{ default (include "mintel_common.fullname" $secretData) .nameOverride }}
```
Comment (barrymars) [inline charts/basic-config/templates/external-secrets.yaml:7]: probably needs a dedicated helper too, as the value will be needed in other places, envFrom, stakater annotations, etc

**Review by barrymars — CHANGES_REQUESTED** [review]

File: charts/basic-config/templates/service.yaml (hunk 2 above)
Comment (nabadger) [reply charts/basic-config/templates/service.yaml]: IK've added a helper `mintel_common.serviceName` and simplified it. I don't know why it was creating a custom `serviceData` dict (the standard-app-stack doesn't do that).

File: charts/basic-config/templates/external-secrets.yaml (hunk 3 above)
Comment (nabadger) [reply charts/basic-config/templates/external-secrets.yaml]: I think this needs more thought as at the moment basic-config does not create any workloads (deployment etc) or Ingress. 

For now I've actually changed this to be consistant with the standard-app-stack.

That is - re-use the `extraSecrets` logic, added more tests, support `nameOverride`.

**Review by barrymars — APPROVED** [review]

**Review by EmmanuelOgiji — APPROVED** [review]

**Pull request merged**